core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp

```diff
@@ -2479,6 +2711,33 @@ static bool
 shift_r_imm_to_r_i32(x86::Assembler &a, SHIFT_OP op, int32 reg_no_dst,
                      int32 reg_no1_src, int32 data2_src)
 {
+    Imm imm(data2_src);
+
+    switch (op) {
+        case SHL:
+        {
+            a.shl(regs_i32[reg_no1_src], imm);
+            break;
+        }
+        case SHRS:
+        {
+            a.sar(regs_i32[reg_no1_src], imm);
+            break;
+        }
+        case SHRU:
+        {
+            a.shr(regs_i32[reg_no1_src], imm);
+            break;
+        }
+        default:
+        {
+            goto fail;
+        }
+    }
+
+    a.mov(regs_i32[reg_no_dst], regs_i32[reg_no1_src]);
```

Comment:
Using `mov_r_to_r(a, JIT_REG_KIND_I32, reg_no_dst, reg_no_src);`, so no need to emit mov instruction when reg_no_dst == reg_no1_src.

Reply:
Just want to make sure. 
Shall we merge all three  `mov_r_to_r` , `mov_r_to_r_i32`, `mov_r_to_r_i64` ?

Reply:
No need to merge, how about using mov_r_ro_r_i32/mov_r_to_r_i64/mov_r_to_r_f32/mov_r_to_r_f64，as them are more concise/simple?
How about changing all mov_r_to_r(..JIT_REG_KIND_I32, ...) to mov_r_to_r_i32() in current code?